climada/util/coordinates.py

```diff
@@ -195,9 +198,11 @@ def lon_bounds(lon, buffer=0.0):
     gap_max = np.argmax(lon_diff)
     lon_diff_max = lon_diff[gap_max]
     if lon_diff_max < 2:
-        # looks like the data covers the whole range [-180, 180] rather evenly
-        lon_min = max(lon_uniq[0] - buffer, -180)
-        lon_max = min(lon_uniq[-2] + buffer, 180)
+        # since the largest gap is comparably small, enforce the [-180, 180] value range
+        gap_max = lon_diff.size - 1
+    if lon_diff_max <= 2 * buffer:
```

Comment:
We are now explicitly handling exceedingly large `buffer` sizes. This fixes a bug in this helper function (see unit test).